src/Microsoft.TemplateEngine.Core/Expressions/VisualBasic/VisualBasicStyleEvaluatorDefintion.cs

```diff
@@ -0,0 +1,357 @@
+                bool r = (bool)Convert.ChangeType(evalResult, typeof(bool));
+                faulted = false;
+                return r;
+            }
+            catch
+            {
+                faulted = true;
+                return false;
+            }
+        }
+
+        public static bool EvaluateFromString(IEngineEnvironmentSettings environmentSettings, string text, IVariableCollection variables)
+        {
+            using (MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(text)))
+            using (MemoryStream res = new MemoryStream())
+            {
+                EngineConfig cfg = new EngineConfig(environmentSettings, variables);
+                IProcessorState state = new ProcessorState(ms, res, (int)ms.Length, (int)ms.Length, cfg, NoOperationProviders);
+                int len = (int)ms.Length;
+                int pos = 0;
+                return Evaluate(state, ref len, ref pos, out bool faulted);
+            }
+        }
+
+        private static int? AttemptBooleanComparison(object left, object right)
+        {
+            bool leftIsBool = Map.TryConvert(left, out bool lb);
+            bool rightIsBool = Map.TryConvert(right, out bool rb);
+
+            if (!leftIsBool || !rightIsBool)
+            {
+                return null;
+            }
+
+            return lb.CompareTo(rb);
+        }
+
+        private static int? AttemptComparableComparison(object left, object right)
+        {
+            IComparable ls = left as IComparable;
+            IComparable rs = right as IComparable;
+
+            if (ls == null || rs == null)
+            {
+                return null;
+            }
+
+            return ls.CompareTo(rs);
+        }
+
+        private static int? AttemptLexographicComparison(object left, object right)
+        {
+            string ls = left as string;
+            string rs = right as string;
+
+            if (ls == null || rs == null)
+            {
+                return null;
+            }
+
+            return string.Compare(ls, rs, StringComparison.OrdinalIgnoreCase);
+        }
+
+        private static int? AttemptNumericComparison(object left, object right)
+        {
+            bool leftIsDouble = Map.TryConvert(left, out double ld);
+            bool rightIsDouble = Map.TryConvert(right, out double rd);
+
+            if (!leftIsDouble)
+            {
+                if (!Map.TryConvert(left, out long ll))
+                {
+                    return null;
+                }
+
+                ld = ll;
+            }
+
+            if (!rightIsDouble)
+            {
+                if (!Map.TryConvert(right, out long rl))
+                {
+                    return null;
+                }
+
+                rd = rl;
+            }
+
+            return ld.CompareTo(rd);
+        }
+
+        private static int? AttemptVersionComparison(object left, object right)
+        {
+            Version lv = left as Version;
```

Comment:
I'm surprised you're using the System.Version class. We've got VersionStringHelpers.cs in Utils to deal with version strings without System.Version